src/main/java/ether/TransactionsManager.java

```diff
@@ -437,308 +487,105 @@ private void processGetAllRequest(MeshID sourceId) {
             }
-            byte[] data = JSON.getErrorResponse(EtherUtility.RES_OPEN_CLIENT_TO_SUPER_PEER,
-                    "Failed to get Ether balance.");
-            sendTransaction(sourceId, data);
             return;
         }
 
-        //Get client Token balance
-        String clientTokenBalance;
+        //Check for In-Channel
+        ImmutablePair<ImmutablePair<byte[], BigInteger>, ImmutablePair<byte[], BigInteger>> bill = null;
         try {
-            clientTokenBalance = EtherClient.getTokenBalance(sourceId.toString(), httpAgent);
-        } catch (IOException e) {
+            bill = meshManager.getTransactionManager().getMostRecentBillFromSender(remotePeerMeshId.getRawUuid());
+        } catch (RightMeshException e){
             if (Settings.DEBUG_INFO) {
-                System.out.println("Failed to get Token balance for: " + sourceId);
+                System.out.println("Failed to get the most recent bill from sender, RightMeshException: "
+                        + e.getMessage());
             }
-            byte[] data = JSON.getErrorResponse(EtherUtility.RES_OPEN_CLIENT_TO_SUPER_PEER,
-                    "Failed to get Token balance.");
-            sendTransaction(sourceId, data);
-            return;
         }
 
-        BigInteger clientNonce = EtherClient.getNonce(sourceId.toString(), httpAgent);
-        if (clientNonce == null) {
-            if (Settings.DEBUG_INFO) {
-                System.out.println("Failed to get nonce for: " + sourceId);
-            }
-            byte[] data = JSON.getErrorResponse(EtherUtility.RES_OPEN_CLIENT_TO_SUPER_PEER,
-                    "Failed to get nonce.");
-            sendTransaction(sourceId, data);
-            return;
-        }
-
-        byte[] data = JSON.sendOpenClientToSpResponse(inChannel, clientEtherBalance, clientTokenBalance, clientNonce);
-        sendTransaction(sourceId, data);
-    }
+        if(bill != null) {
+            //In-Channel exist, lets try to close it.
+            System.out.println(ownMeshId + " --> " + remotePeerAddress
+                    + "bill from sender found, trying to close In-Channel...");
 
+            ImmutablePair<byte[], BigInteger> balanceProofSig = bill.getLeft();
+            ImmutablePair<byte[], BigInteger> closingSig = bill.getRight();
 
-    /**
-     * Processes the Balance Message Signature from Client-Remote Peer.
-     *
-     * @param sourceId The MeshID of the remote peer, the sender.
-     * @param jsonObject The transaction data.
-     */
-    private void processBalanceMessageSignatureRequest(MeshID sourceId, JSONObject jsonObject) {
-
-        //Get the Client-->SuperPeer channel object
-        EtherUtility.PaymentChannel inChannel = getChannelFromMap(sourceId.toString(), ownMeshId.toString());
+            //validate the balance in both pairs (balanceProofSig and closingSig)
+            if(balanceProofSig.right.equals(closingSig.right)) {
```

Comment:
For an in-channel, if the balance values are not the same between balanceProof and closingHash, the super peer can recreate a new closingHash signature with the balance in balanceProof, then the channel can be closed.

Reply:
We cannot guarantee the balances are the same, since DATA and DATA_ACK may be in the transmission path.

Reply:
Similar to the in-channel case, when the super peer wants close an out-channel, if the two balances are not the same, the super peer need to recreate a balanceProof signature to match the balance in closingHash. Then, the channel can also be closed. You can keep a log in the output but the channel needs to be closed.

Reply:
done